fairscale/optim/oss.py

```diff
@@ -101,12 +104,17 @@ def __init__(
 
         # Current default device is set by the parameters allocated to this rank
         self._device = list(self.per_device_params.keys())[0]
+        self.work_handles: Deque[Workhandle] = deque()
         self.buckets: Dict[torch.device, List[torch.Tensor]] = {}
-        self.buffer_max_size = broadcast_buffer_size
+        self._setup_flat_buffers()
 
-        self.should_bucket_param: List[bool] = []
-        self.work_handles: Deque[Workhandle] = deque()
-        self._setup_bucket_strategy()
+    def refresh_trainability(self) -> None:
```

Comment:
Open question @msbaines and @min-xu-ai , I just added these two lines (and a warning in the class docstring), related to https://github.com/facebookresearch/fairscale/pull/369, usecase which popped up for VISSL: freeze/unfreeze some parameters on the fly.
- what do you think of this option (if you changed the graph, call this refresh function, opt-in basically) ? still ok for you ? Not landing before a reaction here :)
- how does Pipe work with that, do you reevalue the graph all the time or is it frozen, or a solution like this one ?

Reply:
removing that, better to keep it for another PR if need be (see discussion on #369)